src/material/form-field/form-field.scss

```diff
@@ -179,10 +179,14 @@ $_icon-prefix-infix-padding: 4px;
   flex: auto;
   min-width: 0;
 
-  // Infix stretches to fit the container, but naturally wants to be this wide. We set
-  // this in order to have a consistent natural size for the various types of controls
-  // that can go in a form field.
-  width: 180px;
+  // We do not to set fixed width on textarea with cols attribute as it makes different columns
+  // attribute look same width.
+  :has(:not(textarea[cols])) {
```

Comment:
This is problematic, because it increases the specificity for the `width`. We can flip it around so `:has(textarea[col])` resets the width.